envoy-control-core/src/main/resources/lua/ingress_rbac_logging.lua

```diff
@@ -8,8 +8,12 @@ function envoy_on_request(handle)
     local request_id = first_header_value_from_list(request_id_header_names, handle)
     local trusted_header_name = handle:metadata():get("trusted_client_identity_header") or ""
     local client_name = ""
+    local trusted_client = false
     if trusted_header_name ~= "" then
         client_name = handle:headers():get(trusted_header_name) or ""
+        if client_name ~= "" then
```

Comment:
How about moving this under the line 21 condition, by "not trusted" client name correction instead of adding another condition here?

Reply:
By `this` I ment trusted_client = true`, missclick

Reply:
You're right, that's seems like a bad idea. Code would be terser but more error-prone.